pkg/client/unversioned/request.go

```diff
@@ -232,13 +234,27 @@ func (r *Request) AbsPath(segments ...string) *Request {
 	if r.err != nil {
 		return r
 	}
+	r.path = absPath(r.serverBase, segments...)
+	return r
+}
+
+func absPath(base string, segments ...string) string {
+	// special case: if base ends in /api then the "real" relative server base is ..
+	if strings.HasSuffix(base, "/api") {
+		base = base[0 : len(base)-4]
+	}
+	// join base+segments
 	if len(segments) == 1 {
 		// preserve any trailing slashes for legacy behavior
-		r.path = segments[0]
-	} else {
-		r.path = path.Join(segments...)
+		p := path.Join(base, segments[0])
+		if strings.HasSuffix(segments[0], "/") && len(p) > 1 {
+			p += "/"
+		}
+		return p
 	}
-	return r
+	x := make([]string, 1, len(segments)+1)
```

Comment:
two `path.Join`s or chained `path.Join(base, path.Join(segments...))` would be easier to read.